assets/js/googlesitekit/site/index.js

```diff
@@ -0,0 +1,210 @@
+ */
+import { get, set } from 'googlesitekit-api';
+import {
+	FETCH_CONNECTION_INFO,
+	FETCH_RESET,
+	INITIALIZE,
+	RESET,
+	RESET_FAILURE,
+	RESET_SUCCESS,
+	RECEIVE_CONNECTION_INFO,
+	RECEIVE_CONNECTION_INFO_FAILED,
+} from './index.private';
+
+export const INITIAL_STATE = {
+	connectionInfo: null,
+	isFetchingConnectionInfo: false,
+	isResetting: false,
+};
+
+export const STORE_NAME = 'core/site';
+
+export const actions = {
+	fetchConnectionInfo() {
+		return {
+			payload: {},
+			type: FETCH_CONNECTION_INFO,
+		};
+	},
+
+	fetchReset() {
+		return {
+			payload: {},
+			type: FETCH_RESET,
+		};
+	},
+
+	initialize() {
+		return {
+			payload: {},
+			type: INITIALIZE,
+		};
+	},
+
+	receiveConnectionInfo( connectionInfo ) {
+		invariant( connectionInfo, 'connectionInfo is required.' );
+
+		return {
+			payload: { connectionInfo },
+			type: RECEIVE_CONNECTION_INFO,
+		};
+	},
+
+	receiveConnectionInfoFailed( error ) {
+		invariant( error, 'error is required.' );
+
+		return {
+			payload: { error },
+			type: RECEIVE_CONNECTION_INFO_FAILED,
+		};
+	},
+
+	*reset() {
+		try {
+			yield actions.fetchReset();
+			yield actions.resetSuccess();
+			yield actions.initialize();
+
+			return {
+				payload: {},
+				type: RESET,
+			};
+		} catch ( err ) {
+			return actions.resetFailure( err );
+		}
+	},
+
+	resetFailure( error ) {
+		invariant( error, 'error is required.' );
+		return {
+			payload: { error },
+			type: RESET_FAILURE,
+		};
+	},
+
+	resetSuccess() {
+		return {
+			payload: {},
+			type: RESET_SUCCESS,
+		};
+	},
+};
+
+export const reducer = ( state = INITIAL_STATE, action = {} ) => {
+	switch ( action.type ) {
+		case FETCH_CONNECTION_INFO: {
+			return {
+				...state,
+				isFetchingConnectionInfo: true,
+			};
+		}
+
+		case FETCH_RESET: {
+			return {
+				...state,
+				isResetting: true,
+			};
+		}
+
+		case INITIALIZE: {
+			return { ...INITIAL_STATE };
+		}
+
+		case RECEIVE_CONNECTION_INFO: {
+			const { connectionInfo } = action.payload;
+
+			return {
+				...state,
+				isFetchingConnectionInfo: false,
+				connectionInfo,
+			};
+		}
+
+		case RECEIVE_CONNECTION_INFO_FAILED: {
+			const { error } = action.payload;
+
+			return {
+				...state,
+				isFetchingConnectionInfo: false,
+				connectionInfo: { error, hasError: true },
+			};
+		}
+
+		case RESET_FAILURE: {
+			const { error } = action.payload;
+			return {
+				...state,
+				resetError: error,
+				isResetting: false,
+			};
+		}
+
+		case RESET_SUCCESS: {
+			return {
+				...state,
+				isResetting: false,
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const selectors = {
+	getConnection: ( state ) => {
+		const { connectionInfo } = state;
+
+		return connectionInfo;
+	},
+
+	isFetchingConnectionInfo: ( state ) => {
+		const { isFetchingConnectionInfo } = state;
+
+		return isFetchingConnectionInfo;
+	},
+
+	isResetting: ( state ) => {
+		const { isResetting } = state;
+
+		return isResetting;
+	},
```

Comment:
Similar like with actions, we should introduce a clear naming convention to differentiate these less public, more internal selectors. Let's prefix each of these (here and in the future) with `fetching...` (e.g. `fetchingConnectionInfo` and `fetchingReset`).

Reply:
I worry about the difference between `fetchingConnectionInfo` and `fetchConnectionInfo`—they're quite close and hard to tell apart, but mean very different thing. Checking on the status of a store with selectors that start with *has*, *is*, etc. strikes me as easier-to-differentiate.

`isFetchingConnectionInfo` strikes me as much clearer compared to `fetchingConnectionInfo`. I know `fetchConnectionInfo` isn't public, but even for our own internal codebase that three-letter difference can be tough to spot.

Reply:
I would strongly vote for more clear public API vs this differentiation. FWIW, one is a selector and the other an action. `is...` should be used for boolean public selectors, while using `fetching...` for whether these API requests are in progress are a good indicator for being internal, just like `fetch...` and `receive...` are for the respective actions.

Reply:
We chatted and clarified this on a call, but for posterity the aim of this selector is actually to be used "publicly" by components and to signify whether a "reset" action is currently in-progress. We decided to be consistent with naming, eg. using `isDoingReset` (`isDoing$ACTION_NAME`) but I clarified that these selectors are important for public usage.

In the future we may want to tweak the prefix we use (`isPerforming` or `is$ACTION_NAMEInProgress`), but as discussed in the call using `fetching` couples the action's asynchronous backend/behaviour to the selector, which places an undo burden on the user to know _how_ the action works. So going with a generic `isDoing` prefix means you can check on status without knowing _how_ an action works in the background.

Reply:
Thanks for summarizing, yeah using the `isDoing` is a generic term that doesn't require knowledge about what exactly a certain action is doing internally to "complete itself".